### PR TITLE
Laboratory Work #1, Koykova Aleksandra - 20FPL3

### PR DESCRIPTION
**Pull request by KoykovaAleksandra**

### CODE REVIEW

**Review by dmitry-uraev — COMMENTED** [review]
its time to implement parser and crawler

File: scrapper.py [hunk 1]
```diff
@@ -44,20 +75,94 @@ def get_search_urls(self):
...
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def _fill_article_with_meta_information(self, article_bs):
+        article_title = article_bs.find('h1').text
+        self.article.title = article_title.strip()
+
+        self.article.author = 'NOT FOUND'
```
Comment (dmitry-uraev) [inline scrapper.py:88]: so you do not have author for sure?

File: scrapper.py [hunk 2]
```diff
@@ -44,20 +75,94 @@ def get_search_urls(self):
...
+        text_content = article_bs.find('div', class_="text-content")
+        divs = text_content.find_all('div', class_="text-block")
+        self.article.text = ''
+        for div in divs:
+            p_tags = div.find_all('p')
+            for p_tag in p_tags:
+                self.article.text += p_tag.text.strip()
+
+    def parse(self):
+        response = requests.get(url=self.article_url)
```
Comment (dmitry-uraev) [inline scrapper.py:110]: are any headers or specific request settings required? did you tried to parse 100 articles?

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
Good job. If you want to hand-over the lab this Friday make sure you have green PR

File: scrapper.py (hunk 1 above)
Comment (KoykovaAleksandra) [reply scrapper.py:88]: Yes, I didn't find any authors in the articles on my resource

File: scrapper.py (hunk 2 above)
Comment (KoykovaAleksandra) [reply scrapper.py]: I added headers, but the code still does not pass the tests on the github (the crawler and parser pass in the PyCharm). And sometimes a connection error occurs((.

**Review by dmitry-uraev — COMMENTED** [review]
Good job. The second lab is ready for assignment

**Pull request closed without merging**